apps/client/src/flogo/app/core/apps.service.ts

```diff
@@ -139,6 +183,49 @@ export class AppDetailService {
     );
   }
 
+  public hasTriggers() {
+    return this.triggersState$.pipe(
+      take(1),
+      map(triggers => {
+        const appHasTriggers = triggers && triggers.length > 0;
+        return !!appHasTriggers;
+      })
+    );
+  }
+
+  public getAvailableShimBuildOptions$(): Observable<ShimBuildOptions[]> {
```

Comment:
we only use suffix `$` to indicate variables that are streams, it's not necessary for functions